### ISSUE TITLE
What  Claim Name in openid tokens is used for representing groups?

### ISSUE THREAD

**Issue by trippleoone** [observatorium/api#291]
We are trying to connect directly to the API with for instance Grafana (but it can be other clients for that matter). When working directly with a user it works by setting the “sub”: value in the token to the name of the user. During authorization the API then maps the username of the logged in user to an entry in the rbac. We use Forward OAuth Identity in Grafana and it works fine.
We do however not know how to make it work with groups. What field in the token should be used for groups (or roles?)? We have tried the most obvious “groups”, for instance: "groups": ["/grouptenant1"]. We have also tried to remove the path for example: "groups": ["grouptenant1"]
As a last resort we tried to add the group membership to the sub field. Since it then becomes an array and the code expects a string it of course is a failure.
So in short: How do we present group membership in the token so that the API can use it for authorization?”
We use keycloak to generate the token.

**Comment by squat** [comment]
Hi @trippleoone,
The administrator must specify the claim to use for the groups in the tenant configuration, e.g.:
```yaml
tenants:
- name: test
  id: uuid
  oidc:
    clientID: id
    clientSecret: secret
    issuerURL: url
    redirectURL: redirect_url
    usernameClaim: email
    groupClaim: group
```

**Comment by trippleoone** [comment]
Hi and thank you for the  quick answer. I did a quick test and it works like a charm!

**Comment by squat** [comment]
:+1: I'll close this issue then :)
